### PR TITLE
Fix storage expansion in pallet section

### LINKED ISSUE #5320: pallet_section not work for storage
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

For the pallet which include many storage, we try to use the pallet_section to put some storage in separate file. but we get the compilation error as "error[E0425]: cannot find value `entries` in this scope".

### Steps to reproduce

source code: https://github.com/opentensor/subtensor/tree/junius-try-storage-macro/pallets/subtensor

steps:
1. git clone https://github.com/opentensor/subtensor
2. cd subtensor
3. git checkout junius-try-storage-macro
4. cd pallets/subtensor
5. cargo build

### PR DESCRIPTION
**Pull request by eagr**
fixes #5320 @sam0x17 @gupnik 

# Description

The issue could be confirmed with the added example. The cause is for macro hygiene, `entries` in the `#( #entries_builder )*` expansion won't be able to reference the `entries` defined outside. The solution here is to allow the reference to be passed into the expansion with closure.

Or we could just switch to the unhygienic span with `quote::quote!` instead such that `entries` will resolve to the "outer" definition.

### CODE REVIEW

File: substrate/frame/support/procedural/src/pallet/expand/storage.rs [hunk 1]
```diff
@@ -425,17 +425,20 @@ pub fn expand_storages(def: &mut Def) -> proc_macro2::TokenStream {
 			Ok(deprecation) => deprecation,
 			Err(e) => return e.into_compile_error(),
 		};
+		// FIXME eagr just use quote::quote!() instead?
```
Comment (bkchr) [inline substrate/frame/support/procedural/src/pallet/expand/storage.rs:428]:
```suggestion
```

File: substrate/frame/support/procedural/examples/section/main.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+use frame_support::pallet_macros::{import_section, pallet_section};
```
Comment (bkchr) [inline substrate/frame/support/procedural/examples/section/main.rs:1]: So this failed to compile before this fix?

Could you please move this as test to `frame-support-test/test`. Maybe we should also put the sections into their own files to be even closer to reality.

Comment (eagr) [reply substrate/frame/support/procedural/examples/section/main.rs:1]: Yep, it complains that "cannot find value `entries` in this scope", just as described in the issue.

**Comment by eagr** [comment]
For the added test, without the fix:

```
test tests/split_ui/pass/split_storage.rs [should pass] ... error
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error[E0425]: cannot find value `entries` in this scope
  --> tests/split_ui/pass/split_storage.rs:5:12
   |
5  |     #[pallet::storage]
   |               ^^^^^^^ not found in this scope
...
36 | #[import_section(storage)]
   | -------------------------- in this macro invocation
   |
   = note: this error originates in the macro `__export_tokens_tt_storage` which comes from the expansion of the macro `frame_support::macro_magic::forward_tokens` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0425]: cannot find value `entries` in this scope
  --> tests/split_ui/pass/split_storage.rs:8:12
   |
8  |     #[pallet::storage]
   |               ^^^^^^^ not found in this scope
...
36 | #[import_section(storage)]
   | -------------------------- in this macro invocation
   |
   = note: this error originates in the macro `__export_tokens_tt_storage` which comes from the expansion of the macro `frame_support::macro_magic::forward_tokens` (in Nightly builds, run with -Z macro-backtrace for more info)
```

File: substrate/frame/support/test/tests/split_ui.rs [hunk 3]
```diff
@@ -32,5 +32,5 @@ fn split_ui() {
 
 	let t = trybuild::TestCases::new();
 	t.compile_fail("tests/split_ui/*.rs");
-	t.pass("tests/split_ui/pass/*.rs");
+	t.pass("tests/split_ui/pass/split_*.rs");
```
Comment (gui1117) [inline substrate/frame/support/test/tests/split_ui.rs:35]: another solution is to put the not compiling module at `pass/storage/mod.rs`.
But both solution are ok to me.

**Review by gui1117 — APPROVED** [review]
I'm surprise the code didn't compile.
New code is good, so it is good to me.

File: substrate/frame/support/test/tests/split_ui.rs (hunk 3 above)
Comment (eagr) [reply substrate/frame/support/test/tests/split_ui.rs:35]: That's better.

The issue is a common mistake I guess and the tests somehow miss this case.

**Review by bkchr — APPROVED** [review]

**Pull request merged**